pkg/cloudproduct/cloudproduct.go

```diff
@@ -33,6 +37,12 @@ type CloudProduct interface {
 	// SyncPodPortsToGameServer runs after a Pod has been assigned to a Node and before we sync
 	// Pod host ports to the GameServer status.
 	SyncPodPortsToGameServer(*agonesv1.GameServer, *corev1.Pod) error
+
+	// ValidateGameServer is called by GameServer.Validate to allow for product specific validation.
+	ValidateGameServer(*agonesv1.GameServer) []metav1.StatusCause
+
+	// NewPortAllocator creates a PortAllocator. c.f. gameservers.NewPortAllocator for parameters.
```

Comment:
What's "c.f." ? See function?

Reply:
"see also", except I screwed it up: https://en.wikipedia.org/wiki/Cf.

Will just change to "see".